packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -457,6 +475,21 @@ function NativeStackViewInner({ state, navigation, descriptors }: Props) {
   );
 }
 
+const handleHeaderHeightChangeEvent = (
+  height: number,
+  cachedHeaderHeight: React.MutableRefObject<number>,
+  animatedHeaderHeight: Animated.Value
+) => {
+  if (cachedHeaderHeight.current !== height) {
+    // Currently, we're setting value by Animated#setValue, because we want to cache animated value.
+    // Also, in React Native 0.72 there was a bug on Fabric causing a large delay between the screen transition,
+    // which should not occur.
+    // TODO: Check if it's possible to replace animated#setValue to Animated#event.
```

Comment:
I think that is quite important. If it doesn't work, leave a note why

Reply:
Sure, I rewrited todo in https://github.com/react-navigation/react-navigation/pull/11663/commits/e8f8bb1099524bc357b7bd1bbaf7d415be7628c9.